builder_test.go

```diff
@@ -44,6 +50,23 @@ func TestDeleteBuilder(t *testing.T) {
 		// Limit
 		{users.Delete().Limit(10),
 			"DELETE FROM `users` LIMIT 10"},
+		// Joins
+		{users.InnerJoin(objects).Delete(),
+			"DELETE FROM `users` INNER JOIN `objects`"},
+		{users.InnerJoin(objects).On(foo.Eq(baz)).Delete(),
+			"DELETE FROM `users` INNER JOIN `objects` ON `users`.`foo` = `objects`.`baz`"},
+		{users.InnerJoin(objects).On(foo.Eq(objects.C("foo"))).Delete(),
+			"DELETE FROM `users` INNER JOIN `objects` ON `users`.`foo` = `objects`.`foo`"},
+		{users.InnerJoin(objects).Using(foo).Delete(),
+			"DELETE FROM `users` INNER JOIN `objects` USING (`foo`)"},
```

Comment:
Is this valid MySQL syntax? I think the `USING` should be before the `INNER JOIN`.
See https://dev.mysql.com/doc/refman/5.7/en/delete.html and https://dev.mysql.com/doc/refman/5.7/en/join.html for the abstract syntax.

Reply:
in this case does the "using" apply to the join, not the delete?

Reply:
I just checked. The part that you're referring to is ok. However, there is a different issue that I had to fix to get the SQL to run. The delete doesn't know which table(s) to delete from. So in this scenario if we wanted to delete only the rows in the users table, then it needs to be:
DELETE `users`  FROM `users` INNER JOIN `objects` USING (`foo`)
It could also be this to delete from both:
DELETE `users`, `objects`  FROM `users` INNER JOIN `objects` USING (`foo`)

Reply:
Ok, so Delete may have a USING and join may have a USING. They're different things.
These two things are equivalent:
`DELETE FROM users USING usersINNER JOIN objects USING (foo)`
`DELETE users FROM users INNER JOIN objects USING (foo)`